Emby.Server.Implementations/Library/LibraryManager.cs

```diff
@@ -649,10 +649,16 @@ private Guid GetNewItemIdInternal(string key, Type type, bool forceCaseInsensiti
                 args.FileSystemChildren = files;
             }
 
-            // Check to see if we should resolve based on our contents
-            if (args.IsDirectory && !ShouldResolvePathContents(args))
+            // Filter content based on ignore rules
+            if (args.IsDirectory)
             {
-                return null;
+                var filtered = args.GetActualFileSystemChildren().ToArray();
+                if (filtered.Length == 0)
+                {
+                    return null;
```

Comment:
this causes the tests to fail. When there are no children present, an empty Folder type BaseItem will get returned as Null instead, breaking other parts of the code